### PR TITLE
=str #16923: Inject identity between SubscriberSource and PublisherSink

### PR DESCRIPTION
**Pull request by drewhk**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala [hunk 1]
```diff
@@ -296,10 +296,21 @@ private[akka] object StreamLayout {
   }
 }
 
+private[stream] class SubscriberSourceVirtualProcessor[T] extends Processor[T, T] {
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala]: may want to make this final

File: akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala [hunk 2]
```diff
@@ -296,10 +296,21 @@ private[akka] object StreamLayout {
...
+  override def onSubscribe(s: Subscription): Unit = subscriber.onSubscribe(s)
+  override def onComplete(): Unit = subscriber.onComplete()
+  override def onNext(t: T): Unit = subscriber.onNext(t)
+}
+
 /**
  * INTERNAL API
  */
-private[stream] class VirtualSubscriber[T](val owner: VirtualPublisher[T]) extends Subscriber[T] {
+private[stream] class PublisherSinkVirtualSubscriber[T](val owner: PublisherSinkVirtualPublisher[T]) extends Subscriber[T] {
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala]: final?

File: akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala [hunk 3]
```diff
@@ -309,7 +320,7 @@ private[stream] class VirtualSubscriber[T](val owner: VirtualPublisher[T]) exten
 /**
  * INTERNAL API
  */
-private[stream] class VirtualPublisher[T]() extends Publisher[T] {
+private[stream] class PublisherSinkVirtualPublisher[T]() extends Publisher[T] {
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala]: final?

**Comment by drewhk** [comment]
PLS BUILD

**Comment by drewhk** [comment]
Fixes #16923

**Pull request merged**